experimental/dispatch_profiler/manifest.py

```diff
@@ -1,9 +1,14 @@
-import enum
-import os.path
-import shutil
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+import enum, os, shutil, pickle
```

Comment:
I think it might be fine here but `pickle` is a library we usually want to avoid due to security concerns.

I'll suggest not to use pickle now (it seems like we don't need to serialize `Manifest` before?) and implements a more secure JSON serialization later.

Reply:
We added serialization and deserialization to communicate generator.py 's output to compile.py and profile.py. genreator.py generates a set of dispatches that are consumed by compile.py and profile.py. I used `pickle` bcoz JSON wasn't working for all the custom datatypes we need serialization/deserialization.

Reply:
I think writing custom JSON serialization/deserialization for the custom datatypes should be the approach we wan to go. IREE benchmark framework has the same problem and it has a decorator-style serializer to solve this problem (https://github.com/openxla/iree/blob/main/build_tools/python/e2e_test_framework/serialization.py).

I still think using pickle is going to create security concerns and the serialized files are also not human readable. We can file a TODO for this and refactor this later.

Reply:
Sounds good. I created a new issue to handle this separately (#13495)